lib/front_desk.rb

```diff
@@ -0,0 +1,116 @@
+require_relative "rooms.rb"
+require_relative "reservations.rb"
+
+
+class Front_Desk
+
+  attr_reader :reservations, :rooms, :block_hold
+
+  def initialize
+    @reservations = []
+    @block_reservations = []
+    @rooms = load_rooms
+  end
+
+  def load_rooms
+    room_number = 0
+    rooms = []
+
+    20.times do
+      room_number += 1
+      rooms << Room.new({
+        :room_number => room_number
+      })
+    end
+    return rooms
+  end
+
+  def reserve_room(room_number,start_date, end_date)
+
```

Comment:
You should be checking here to see if the start_date comes before the end date! Not having this fundamentally breaks a lot of the other code you've written!